.github/workflows/module-test-run.yml

```diff
@@ -56,7 +51,7 @@ jobs:
 
             echo -n "testing $name ... "
 
-            if output=$(sudo ./moduletest $recipe --bin ../bin); then
+            if output=$(sudo ./moduletest $recipe --bin ../bin --config ../../src/adapters/pnp/daemon/osconfig.json ); then
```

Comment:
>../bin --config ../../src/adapters/pnp/daemon/osconfig.json [](http://example.com/codeflow?start=56&length=59)

Why?

moduletest supports command line arguments, if invoked manually. We do not need more for now.

Can you please explain why do you believe this change is necessary and what needs is going to solve?

Thanks

Reply:
Allowing moduletest to get configuration file as argument we can optimize module-test-run by eliminating duplicate copy.
This allows to:
- Test with different configurations without necessary copy
- Allow to run moduletest without need access to /etc file system

So instead of  flow
cp ./src/adapters/pnp/daemon/osconfig.json  /etc/osconfig/osconfig.json 
$EDIT /etc/osconfig/osconfig.json (may require root)
./moduletest $recipe --bin ../bin

you can just do

$EDIT  ./src/adapters/pnp/daemon/osconfig.json
./moduletest $recipe --bin ../bin --config ../../src/adapters/pnp/daemon/osconfig.json

Moreover you can have multiple  ./src/adapters/pnp/daemon/osconfig.json and you can juggle them using cmdline switch not cp  ./src/adapters/pnp/daemon/osconfig.json.1 /etc.osconfig/osconfig.json  and remember which one you tested with.

Reply:
There is no real motivation for this change other than maybe not have to run elevated the tests. The tests need to be run elevated regardless of this file (which again, is the global OSConfig configuration file, is not test configuration or anything of that sort) because it's functional tests that mimmick OSConfig usage of the modules under test (e.g. the Security Baseline module must be executed from elevated process, etc).

This PR should be recalled.

Reply:
Even the title of the PR is misleading - that file is NOT the configuration of the test....